drivers/serial/uart_nrfx_uarte2.c

```diff
@@ -0,0 +1,1031 @@
+
+	err = nrfx_uarte_rx_enable(instance, 0);
+	__ASSERT_NO_MSG(err == NRFX_SUCCESS || err == NRFX_ERROR_BUSY);
+
+	(void)err;
+
+	return 0;
+}
+
+static void async_to_irq_trampoline(const struct device *dev)
+{
+	const struct uarte_nrfx_config *cfg = dev->config;
+	struct uarte_nrfx_data *data = dev->data;
+	uint32_t prev = atomic_or(&data->flags, UARTE_DATA_FLAG_TRAMPOLINE);
+
+	if (!(prev & UARTE_DATA_FLAG_TRAMPOLINE)) {
+		nrfx_uarte_int_trigger(&cfg->nrfx_dev);
+	}
+}
+
+static int uarte_nrfx_init(const struct device *dev)
+{
+	int err;
+	nrfx_err_t nerr;
+	const nrfx_uarte_t *nrfx_dev = get_nrfx_dev(dev);
+	const struct uarte_nrfx_config *cfg = dev->config;
+	struct uarte_nrfx_data *data = dev->data;
+
+	err = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
+	if (err < 0) {
+		return err;
+	}
+
+	if (UARTE_ENHANCED_POLL_OUT && cfg->nrfx_config.tx_stop_on_end) {
+		err = endtx_stoptx_ppi_init(nrfx_dev->p_reg);
+		if (err < 0) {
+			return err;
+		}
+	}
+
+	if (UARTE_ANY_INTERRUPT_DRIVEN) {
+		if (cfg->a2i_config) {
+			err = uart_async_to_irq_init(data->a2i_data, cfg->a2i_config);
+			if (err < 0) {
+				return err;
+			}
+		}
+	}
+
+	if (IS_ENABLED(UARTE_INT_ASYNC) && data->async) {
+		k_timer_init(&data->async->rx_timer, rx_timeout_handler, NULL);
+		k_timer_user_data_set(&data->async->rx_timer, (void *)dev);
+		k_timer_init(&data->async->tx_timer, tx_timeout_handler, NULL);
+		k_timer_user_data_set(&data->async->tx_timer, (void *)dev);
+	}
+
+	nerr = nrfx_uarte_init(nrfx_dev, &cfg->nrfx_config,
+			IS_ENABLED(UARTE_INT_ASYNC) ?
+				(IS_POLLING_API(dev) ? NULL : evt_handler) : NULL);
+	if (nerr == NRFX_SUCCESS && !IS_ASYNC_API(dev) && !(cfg->flags & UARTE_CFG_FLAG_NO_RX)) {
+		err = start_rx(dev);
+	}
+
+	switch (nerr) {
+	case NRFX_ERROR_INVALID_STATE:
+		return -EBUSY;
+	case NRFX_ERROR_BUSY:
+		return -EACCES;
+	case NRFX_ERROR_INVALID_PARAM:
+		return -EINVAL;
+	default:
+		return 0;
+	}
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int stop_rx(const struct device *dev)
+{
+	const struct uarte_nrfx_config *cfg = dev->config;
+
+	if (IS_INT_DRIVEN_API(dev)) {
+		return uart_async_to_irq_rx_disable(dev);
+	}
+
+	__ASSERT_NO_MSG(IS_POLLING_API(dev));
+	nrfx_err_t err;
+	const nrfx_uarte_t *instance = &cfg->nrfx_dev;
+
+	err = nrfx_uarte_rx_abort(instance, true, true);
+	__ASSERT_NO_MSG(err == NRFX_SUCCESS);
+
+	return 0;
+}
+
+static int uarte_nrfx_pm_action(const struct device *dev,
+				enum pm_device_action action)
+{
+	const struct uarte_nrfx_config *cfg = dev->config;
+	int ret;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		if (cfg->flags & UARTE_CFG_FLAG_GPIO_MGMT) {
+			ret = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
+			if (ret < 0) {
+				return ret;
+			}
+		}
+		if (!IS_ASYNC_API(dev) && !(cfg->flags & UARTE_CFG_FLAG_NO_RX)) {
+			return start_rx(dev);
+		}
+
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		if ((IS_INT_DRIVEN_API(dev) || IS_POLLING_API(dev)) &&
+		     !(cfg->flags & UARTE_CFG_FLAG_NO_RX)) {
```

Comment:
For consistency with `PM_DEVICE_ACTION_RESUME`:
```suggestion
		if (!IS_ASYNC_API(dev) && !(cfg->flags & UARTE_CFG_FLAG_NO_RX)) {
```